poetry_to_pre_commit/sync_hooks_additional_dependencies.py

```diff
@@ -68,12 +74,39 @@ def get_poetry_deps(*, cwd: pathlib.Path | None = None, group: str) -> Iterable[
         yield f"{dep.complete_name}=={package.version}"
 
 
+def update_or_remove_additional_deps(
+    poetry_deps: set[str], hook_additional_deps: list[str]
+) -> set[str]:
+    # Additional packages that are already in pre-commit configuration could be listed with
+    # any format that is accepted by pip. The following regex might not cover all the cases.
+    current_deps = re.findall(
+        r"^\s*([^=><\[\s]+)", "\n".join(hook_additional_deps), re.MULTILINE
+    )
```

Comment:
I believe [`packaging.requirements.Requirement`](https://packaging.pypa.io/en/stable/requirements.html#requirements) is a much more robust solution. This project is partially born from not wanting to use regex parsing for pre-commit config, so I'm much more in favor of proper parsing, even it if means one more dependency.

Reply:
Done - much better! Also this was already an indirect dependency, so we basically got it for free :-)